packages/workspace/src/browser/workspace-service.ts

```diff
@@ -437,7 +435,9 @@ export class WorkspaceService implements FrontendApplicationContribution {
     }
 
     async getUntitledWorkspace(): Promise<URI> {
-        return getTemporaryWorkspaceFileUri(this.envVariableServer);
+        const configDirURI = await this.envVariableServer.getConfigDirUri();
+        const semiRandomID = (Date.now() + Math.floor(Math.random() * 1000)).toString();
+        return new URI(configDirURI).resolve(`workspaces/Untitled-${semiRandomID}.${THEIA_EXT}`);
```

Comment:
Suggestion: Instead of generating a random ID, we could first generate an URI for `Untitled.theia`, check whether that already exists, increment, generate `Untitled-1`, etc. until we arrive at a number, which does not exist. I think that would be preferable, but takes a bit more time, especially if we have a large amount of untitled workspaces. (unlikely?) What do you think?

Reply:
That's doable. The main risk is in the browser, where we can't guarantee that we'll actually get to run our cleanup code. In that context, people could generate a lot of untitled workspaces without realizing it. But in any case, whether the suffix is random or ordinal, I think it's a good idea to check for existence before saving in any case. Even a small chance of clobbering someone's active workspace should be avoided.

Reply:
I decided to leave the random suffix, because the ordinal approach is guaranteed to get linearly worse if the files don't get cleaned up, whereas the random approach will get only probabilistically worse with each file, but I have added code that checks for existence before returning the URI. I've also added code that cleans up a now-unused untitled file when the 'save workspace as' command is run.